interfaces/repo.go

```diff
@@ -808,6 +808,9 @@ func (r *Repository) SnapSpecification(securitySystem SecuritySystem, snapName s
 // Unknown interfaces and plugs/slots that don't validate are not added.
 // Information about those failures are returned to the caller.
 func (r *Repository) AddSnap(snapInfo *snap.Info) error {
+	if snapInfo.Broken != "" {
```

Comment:
My gut feeling is that this is the key, having this error and the logging about it. Thanks for adding this!

Reply:
Yes, I was surprised to make this connection. I had the check in ifacemgr earlier in this branch but this is where it belongs, I think.